### PR TITLE
make `decodeFunctionCall` and `decodeFunctionReturn` available at `web3-eth-abi`

### LINKED ISSUE #4917: Add decode function parameters
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Request

https://github.com/ChainSafe/web3.js/issues/3016

See here https://github.com/ChainSafe/web3.js/issues/3016#issuecomment-828274902

In reference to the this problem people have, decoding a function requires removing the first 8 characters after the 0x. 

Adding a decodeFunctionParameters method would solve this, simply by splicing the first 8 characters after the 0x and returning the parameters. 

### Use Cases

Used in getting the returned parameter data of function calls. 

Stops people from being confused about why decodeParameters is not working. 

### Implementation Ideas

_No response_

### Feature Examples/References

Ugly example, but it works to prove the point.

```
let web3 = new Web3('https://bsc-dataseed.binance.org/');
		const a = await web3.eth.getTransaction('0x041e65ab1b3ef192d1e2ae38d167f6293f026b20da980f4987e173c9ffcfc886');
		let find = a.input.slice(2, 10); <--- Slice the 8 characters out of the hex
		let remove = a.input.replace(find, ''); <--- Remove the 8 characters
		const z = await web3.eth.abi.decodeParameters(
			[
				{
					type: 'address',
					name: 'token',
				},
				{
					type: 'uint256',
					name: 'amount',
				},
				{
					type: 'address',
					name: 'receiverAddress',
				},
				{
					type: 'uint256',
					name: 'targetChainId',
				}
			],
			remove <--- Hex with function ID removed
		);
```

```
let remove = `0x${a.input.substr(10)}`; <---- Remove the function ID, add back in the 0x
		const z = await web3.eth.abi.decodeParameters(
			[
				{
					type: 'address',
					name: 'token',
				},
				{
					type: 'uint256',
					name: 'amount',
				},
				{
					type: 'address',
					name: 'receiverAddress',
				},
				{
					type: 'uint256',
					name: 'targetChainId',
				}
			],
			remove
		);
```

`web3.eth.abi.decodeFunctionParameters([<PARAMETERS>], string)`

Final output is CORRECT data.

### Are you willing to implement this feature?

- [x] Yes

### PR DESCRIPTION
**Pull request by Muhammad-Altabba**
## Description

Fixes #4917 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run lint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:coverage` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have linked Issue(s) with this PR in "Linked Issues" menu.

### CODE REVIEW

File: packages/web3-eth-contract/src/contract.ts [hunk 1]
```diff
@@ -1026,7 +1023,7 @@ export class Contract<Abi extends ContractAbi>
 				`The ABI for the provided method signature ${methodSignature} was not found.`,
 			);
 		}
-		return { ...decodeMethodParams(abi, data), __method__: jsonInterfaceMethodToString(abi) };
```
Comment (Muhammad-Altabba) [inline packages/web3-eth-contract/src/contract.ts:1029]: the property `__method__` is now returned by `decodeMethodParams`.

**Review by avkos — APPROVED** [review]

File: packages/web3-eth-abi/src/api/functions_api.ts [hunk 2]
```diff
@@ -143,3 +143,67 @@ export const encodeFunctionCall = (
...
+		methodSignatureProvided && data && data.length >= 10 && data.startsWith('0x')
+			? data.slice(10)
+			: data;
+	if (!functionsAbi.inputs) {
+		throw new Web3ContractError('No inputs found in the ABI');
+	}
+	const result = decodeParameters([...functionsAbi.inputs], value);
+	return {
+		...result,
+		__method__: jsonInterfaceMethodToString(functionsAbi),
```
Comment (jdevcs) [inline packages/web3-eth-abi/src/api/functions_api.ts:170]:
```suggestion
		method: jsonInterfaceMethodToString(functionsAbi),
```
why __method__ is used ?

File: packages/web3-eth-abi/src/api/functions_api.ts [hunk 3]
```diff
@@ -143,3 +143,67 @@ export const encodeFunctionCall = (
...
 	).replace('0x', '')}`;
 };
+
+/**
+ * Decodes a function call data using its `JSON interface` object.
+ * The JSON interface spec documentation https://docs.soliditylang.org/en/latest/abi-spec.html#json
+ * @param functionsAbi - The `JSON interface` object of the function.
+ * @param data - The data to decode
+ * @param methodSignatureProvided - (Optional) if `false` do not remove the first 4 bytes that would rather contain the function signature.
+ * @returns - The data decoded according to the passed ABI.
```
Comment (jdevcs) [inline packages/web3-eth-abi/src/api/functions_api.ts:153]: could you add example

File: packages/web3-eth-abi/src/api/functions_api.ts [hunk 4]
```diff
@@ -143,3 +143,67 @@ export const encodeFunctionCall = (
...
+) => {
+	// If it is a constructor there is nothing to decode!
+	if (functionsAbi.type === 'constructor') {
+		return returnValues;
+	}
+
+	if (!returnValues) {
+		// Using "null" value intentionally to match legacy behavior
+		// eslint-disable-next-line no-null/no-null
+		return null;
```
Comment (jdevcs) [inline packages/web3-eth-abi/src/api/functions_api.ts:269]: following lines needs unit tests coverage,

**Review by jdevcs — COMMENTED** [review]
LGTM, only need some docs and unit tests before merging. thanks

**Pull request merged**